discovery-client/src/main/java/io/micronaut/discovery/consul/client/v1/ConsulOperations.java

```diff
@@ -48,33 +49,70 @@ public interface ConsulOperations {
     Publisher<Boolean> putValue(String key, @Body String value);
 
     /**
-     * Reads a Key from Consul. See https://www.consul.io/api/kv.html.
+     * Reads a Key from Consul. It will return all KV matching key as prefix.
+     * See <a href="https://www.consul.io/api/kv.html">https://www.consul.io/api/kv.html</a>.
      *
-     * @param key The key to read
+     * @param key The prefix to match
      * @return A {@link Publisher} that emits a list of {@link KeyValue}
+     * @see #readValues(String, Boolean)
      */
-    @Get(uri = "/kv/{+key}?recurse", single = true)
-    Publisher<List<KeyValue>> readValues(String key);
+    default Publisher<List<KeyValue>> readValues(String key) {
+        return readValues(key, true);
+    }
 
     /**
-     * Reads a Key from Consul. See https://www.consul.io/api/kv.html.
+     * Reads a Key from Consul. See <a href="https://www.consul.io/api/kv.html">https://www.consul.io/api/kv.html</a>.
      *
-     * @param key        The key
+     * @param key     The key or prefix to read
+     * @param recurse Specifies if the lookup should be recursive and treat key as a prefix instead of a literal match.
+     * @return A {@link Publisher} that emits a list of {@link KeyValue}
+     * @see #watchValues(String, Boolean, Integer)
+     */
+    default Publisher<List<KeyValue>> readValues(String key,
+                                                 @Nullable Boolean recurse) {
+        return watchValues(key, recurse, null);
+    }
+
+    /**
+     * Watch a Key from Consul. See <a href="https://www.consul.io/api/kv.html">https://www.consul.io/api/kv.html</a>.
+     *
+     * @param key     The key or prefix to watch
+     * @param recurse Specifies if the lookup should be recursive and treat key as a prefix instead of a literal match.
+     * @param index   The current index if the resource to watch. When provided,
+     *                the call be a <a href="https://developer.hashicorp.com/consul/api-docs/features/blocking">Blocking query</a>
+     *                hanging until a change of the watched resources or timeout is reached.
+     * @return A {@link Publisher} that emits a list of {@link KeyValue}
+     */
+    @Get(uri = "/kv/{+key}?{&recurse}{&index}", single = true)
+    @Retryable(
+        attempts = AbstractConsulClient.EXPR_CONSUL_CONFIG_RETRY_COUNT,
+        delay = AbstractConsulClient.EXPR_CONSUL_CONFIG_RETRY_DELAY
+    )
+    Publisher<List<KeyValue>> watchValues(String key,
```

Comment:
I haven't found how to set dynamically a different timeout for this method when used in a blocked query.
Maybe it would be better to have it in a separate client, to be able to set its own configuration ?